src/transaction/log_page_buffer.c

```diff
@@ -1123,95 +807,64 @@ logpb_fix_page (THREAD_ENTRY * thread_p, LOG_PAGEID pageid, PAGE_FETCH_MODE fetc
   assert ((fetch_mode == NEW_PAGE) || (fetch_mode == OLD_PAGE));
   assert (LOG_CS_OWN_WRITE_MODE (thread_p));
 
-  START_EXCLUSIVE_ACCESS_LOG_PB (rv, thread_p);
 
-  log_bufptr = (LOG_BUFFER *) mht_get (log_Pb.ht, &pageid);
-  if (log_bufptr == NULL)
+  if (pageid == LOGPB_HEADER_PAGE_ID)
     {
-      /* 
-       * Page is not resident. Find a replacement buffer for it
-       */
-
-      stat_page_found = PERF_PAGE_MODE_OLD_LOCK_WAIT;
-
-      while (1)
-	{
-	  log_bufptr = logpb_replace (thread_p, &retry);
-	  if (log_bufptr == NULL)
-	    {
-	      if (retry)
-		{
-		  continue;
-		}
-	      else
-		{
-		  goto error;
-		}
-	    }
-	  else
-	    {
-	      break;
-	    }
-	}
-
-      assert (log_bufptr != NULL);
-
-      /* 
-       * Remove the current page from the hash table and get the desired
-       * page from disk when the page is an old page
-       */
-      if (log_bufptr->pageid != NULL_PAGEID)
-	{
-	  (void) mht_rem (log_Pb.ht, &log_bufptr->pageid, NULL, NULL);
-	}
-
-      /* Fix the page and mark its pageid invalid */
-      log_bufptr->pageid = NULL_PAGEID;
-      log_bufptr->fcnt++;
-
-      phy_pageid = logpb_to_physical_pageid (pageid);
-
-      if (fetch_mode == NEW_PAGE)
+      log_bufptr = &log_Pb.header_buffer;
+    }
+  else
+    {
+      log_bufptr = &log_Pb.buffers[log_get_log_buffer_index ((int) pageid)];
+    }
+  if (fetch_mode == OLD_PAGE)
+    {
+      if (log_bufptr->pageid == NULL_PAGEID || (log_bufptr->pageid != NULL_PAGEID && log_bufptr->pageid != pageid))
 	{
-	  log_bufptr->logpage.hdr.logical_pageid = pageid;
-	  log_bufptr->logpage.hdr.offset = NULL_OFFSET;
+	  log_bufptr->fcnt++;
```

Comment:
@funkydvd 

I'm not quite sure we can just replace the existing buffer. Please imagine a case that the existing page is dirty. 
You can refer to the existing code of `logpb_replace` to handle a dirty log buffer:
```C
                  if (log_bufptr->dirty == true)
                    {
                      END_EXCLUSIVE_ACCESS_LOG_PB (rv, thread_p);

                      assert (LOG_CS_OWN_WRITE_MODE (thread_p));
                      log_Stat.log_buffer_flush_count_by_replacement++;
                      logpb_flush_all_append_pages (thread_p);

                      START_EXCLUSIVE_ACCESS_LOG_PB (rv, thread_p);
                      mnt_log_replacements (thread_p);
                      *retry = true;
                      return NULL;
                    }
```